ivy/func_wrapper.py

```diff
@@ -567,6 +569,68 @@ def _wrapped(func):
     return _wrapper_outer
 
 
+# nans Handling #
+# --------------#
+
+
+def _has_nans(x):
```

Comment:
for maximal generalisation, `_has_nans` should make use `ivy.nested_any`, such as:

```
def _leaf_has_nans(x):
    if isinstance(x, ivy.Container):
        return x.has_nans()
    elif ivy.is_array(x):
        return ivy.isnan(x).any()
    elif x is float("nan"):
        return True
    return False

def _nest_has_nans(x):
    return ivy.nested_any(x, _leaf_has_nans)
```

Reply:
Makes sense! Added in https://github.com/unifyai/ivy/pull/6748/commits/130b1ef83c5f6b13681370271c0b9cb767993003